src/main/11-iam.tf

```diff
@@ -1,4 +1,32 @@
 data "aws_iam_policy_document" "registry_reader_policy" {
+  statement {
+    sid    = "XRayIntegration"
+    effect = "Allow"
+    actions = [
+      "xray:GetGroups",
+      "xray:GetSamplingStatisticSummaries",
+      "xray:PutTelemetryRecords",
+      "xray:GetTraceGraph",
+      "xray:GetServiceGraph",
+      "xray:GetInsightImpactGraph",
+      "xray:GetInsightSummaries",
+      "xray:GetSamplingTargets",
+      "xray:PutTraceSegments",
+      "xray:BatchGetTraces",
+      "xray:BatchGetTraceSummaryById",
+      "xray:GetTimeSeriesServiceStatistics",
+      "xray:GetEncryptionConfig",
+      "xray:GetSamplingRules",
+      "xray:GetInsight",
+      "xray:GetDistinctTraceGraphs",
+      "xray:GetInsightEvents",
+      "xray:GetTraceSummaries"
+    ]
+
+    resources = [
+      "*"
+    ]
+  }
```

Comment:
Does the pod need all of these permissions? The `AWSXRayDaemonWriteAccess` AWS managed policy should be enough. You can apply it directly on the pod role instead of duplicating these permissions in each policy.